03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
-objects on the earth. In comparison, a `Digital Terrain Model (DTM)` or `Digital Elevation
-Model (DEM)` contains elevation data for the actual ground (with trees, buildings and 
-other objects removed). 
-
-In ecology, we are often interested in measuring the heights of trees and so the
-raster that we really want is the `difference` between the `DSM` and the `DTM`.
-This data product is often referred to as a `Canopy Height Model (CHM)` and represents
-the actual height of trees, buildings etc above the ground.
-
-We can calculate this difference by subtracting the two rasters in `R`. However
-if our rasters get large, we can calculate the difference more efficiently using
-the `overlay` function. 
-
-First, let's create and plot a `CHM` by subtracting two rasters - we often call
-this `raster math` in the geospatial world.
-
+As introduced in Lesson 02, we often want to overlay two or more rasters on top 
+of each other, and perform calculations to create a new output raster. If 
+interested in measuring the heights of trees, then what we really want is the 
+`difference` between the Digital Surface Model (`DSM`, tops of trees) and the 
+Digital Terrain Model (`DTM`, ground level) .This data product is often referred
+to as a Canopy Height Model (`CHM`) and represents the actual height of trees, 
+buildings, etc above the ground.
+
+We can calculate this difference in two different ways:
+1) by subtracting the two rasters in `R` 
+or if our rasters are large
+2) using the `overlay()` function to more efficiently calculate the difference.
+
+First, we'll learn the intuitive first step of subtracting layers to create and
+plot a `CHM`.  In the geospatial world we often call
+this *raster math*.
+
+If you completed Lessons 00 and 02 you've already loaded and viewed 
+`HARV_dtmCrop.tif` and `HARV_dsmCrop.tif` and can skip this code chunk.  If you 
+have not completed it, let's load and check out our raster files. 
 ```{r load-libraries }
-
 #load raster package
 library(raster)
 
-#view info about the dtm raster data
+#view info about the dtm & dsm raster data
 GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#load the DTM
+GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dsmCrop.tif")
+#load the DTM & DSM rasters
 DTM <- raster("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#create a quick plot
+DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
+#create a quick plot of each to see what we're dealing with
 plot(DTM,
      main="Digital Terrain Model (Elevation)\n Harvard Forest")
 
-#If it isn't already loaded, load the DSM as well 
-DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
-
+plot(DSM,
+     main="Digital Surface Model (Elevation)\n Harvard Forest")
 ```
 
 Once the data are loaded, we can use mathametical operations on them.
-For instance, we can subtract the DTM from the DSM to create a `Canopy Height
-Model`.
+For instance, we can subtract the DTM from the DSM to create a Canopy Height
+Model.
```

Comment:
Hey - what are your thoughts on this @mjones01 . What the others suggested is we JUST teach one. so the raster math becomes an example rather than a part of the lesson. Thoughts? We want things to be simple - or as simple as possible but i wanted people to see there are different approaches - overlay is just more efficient and thus better.